quickstart/101-vm-with-infrastructure/main.tf

```diff
@@ -116,13 +110,12 @@ resource "azurerm_linux_virtual_machine" "my_terraform_vm" {
     version   = "latest"
   }
 
-  computer_name                   = "myvm"
-  admin_username                  = "azureuser"
-  disable_password_authentication = true
+  computer_name                   = "hostname"
+  admin_username                  = coalesce(var.username, "azureuser")
```

Comment:
Since we have set [default value](https://github.com/Azure/terraform/pull/238/files#diff-effdf19b131d91c87ba12e37d2e50eb8382e761eb9013c4aed10778debdfe6d6R17) for `var.username` already I think we can skip `coalesce` here and use `var.username` directly.

Reply:
Using the coalesce() function allows us to give the reader two options: variable or random. If the reader blanks out the var, they get a random value - such as if they're testing the code by running it multiple times and can't use the same value twice.